rflx/pyrflx/typevalue.py

```diff
@@ -1356,6 +1348,11 @@ def subst(expression: Expr) -> Expr:
             if expression in self._parameters:
                 assert isinstance(expression, Name)
                 return self._parameters[expression]
+            if isinstance(expression, Variable) and expression.name in self.fields:
+                if self._fields[expression.identifier.flat].set:
+                    exp_value = self._fields[expression.identifier.flat].typeval.value
+                    assert isinstance(exp_value, bytes)
+                    return Number(int.from_bytes(exp_value, "big"))
```

Comment:
Does this also work correctly when using `Byte_Order => Low_Order_First`?

Reply:
I guess not generally, but the test case does also succeed when adding this. 
Where would I check if `Byte_Order => Low_Order_First` is set?

Reply:
The byte order should not be relevant when comparing opaque fields with aggregates. The issue should vanish when converting `bytes` to `Aggregate` instead of `Number` (see also other comment regarding comparing aggregates and number).